bluebrain/repo-patches/packages/neuron/package.py

```diff
@@ -297,6 +297,12 @@ def cmake_enable_option(spec_requiremement):
             compilation_flags.append(
                 self.spec.architecture.target.optimization_flags(self.spec.compiler)
             )
+            # In case we're using GCC compiler we enable `-ffast-math` to allow vectorization
+            # of mechanism kernels in case `libmvec` is available in the system.
+            # Due to the fact that the generated code by NMODL includes `#pragma omp simd` clauses
```

Comment:
Should we enable `openmp` variable by default?

Reply:
there is the `openmp-simd` flag maybe by default when building gcc we can just enable that - not to require to omuch?

Reply:
Sorry I didn't get what you mean by 
> `openmp-simd` flag
You mean enabling `openmp` variant in spack when we have `GCC` compiler and `Release/RelWithDebInfo` build by default?

Reply:
I would be happier enabling the bare minimum subset of `-ffast-math` :grimacing:; https://gcc.gnu.org/onlinedocs/gcc-12.3.0/gcc/Optimize-Options.html#index-ffast-math defines what it includes.

I think @ohm314 is referring to https://gcc.gnu.org/onlinedocs/gcc-12.3.0/gcc/C-Dialect-Options.html#index-fopenmp-simd, which is I think a good suggestion: we can pass `-fopenmp-simd` when the `openmp` variant **is not** set, and `-fopenmp` when it **is** set , plus the minimum subset of -ffast-math assuming that's not too scary...

Reply:
> I would be happier enabling the bare minimum subset of -ffast-math grimacing; https://gcc.gnu.org/onlinedocs/gcc-12.3.0/gcc/Optimize-Options.html#index-ffast-math defines what it includes.

I guess this makes sense. With `Intel/oneAPI` compilers in release builds there is by default `-fp-model=fast` and I'm not entirely sure what is enabled there..

> I think @ohm314 is referring to https://gcc.gnu.org/onlinedocs/gcc-12.3.0/gcc/C-Dialect-Options.html#index-fopenmp-simd

I see.. I couldn't find this by googling :see_no_evil: I think it makes sense

Reply:
I would not hold up Intel's defaults as a model of _good_ defaults 😬

For the oneAPI compiler, their documentation [seems a little vague](https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/developer-guide-reference/2023-1/fp-model-fp.html), but empirically then even just passing `-O1` enables things like assuming floating point associativity.

For GCC, the "best" subset I came up with that doesn't stop it vectorising the example is `-funsafe-math-optimizations -fno-math-errno -fno-associative-math -ffinite-math-only` where everything except `-fno-associative-math` seems to be required, and `-fno-associative-math` is because I really don't want to enable that globally in NEURON.